internal/storage/caching/fast_stat_bucket_test.go

```diff
@@ -135,6 +135,80 @@ func (t *CreateObjectTest) WrappedSucceeds() {
 	ExpectEq(obj, o)
 }
 
+////////////////////////////////////////////////////////////////////////
+// CreateChunkUploader
+////////////////////////////////////////////////////////////////////////
+
+type CreateChunkUploaderTest struct {
+	fastStatBucketTest
+}
+
+func init() { RegisterTestSuite(&CreateChunkUploaderTest{}) }
+
+func (t *CreateChunkUploaderTest) CallsEraseAndWrapped() {
+	const name = "taco"
+
+	// Erase
+	ExpectCall(t.cache, "Erase")(name)
+
+	// Wrapped
+	var wrappedReq *gcs.CreateObjectRequest
+	ExpectCall(t.wrapped, "CreateChunkUploader")(Any(), Any(), Any(), Any()).
+		WillOnce(DoAll(SaveArg(1, &wrappedReq), Return(nil, errors.New(""))))
+
+	// Call
+	req := &gcs.CreateObjectRequest{
+		Name: name,
+	}
+
+	_, _ = t.bucket.CreateChunkUploader(context.TODO(), req, 1, nil)
+
+	AssertNe(nil, wrappedReq)
+	ExpectEq(req, wrappedReq)
+}
+
+func (t *CreateChunkUploaderTest) WrappedFails() {
+	var err error
+
+	// Erase
+	ExpectCall(t.cache, "Erase")(Any())
+
+	// Wrapped
+	ExpectCall(t.wrapped, "CreateChunkUploader")(Any(), Any(), Any(), Any()).
+		WillOnce(Return(nil, errors.New("taco")))
+
+	// Call
+	_, err = t.bucket.CreateChunkUploader(context.TODO(), &gcs.CreateObjectRequest{}, 1, nil)
+
+	ExpectThat(err, Error(HasSubstr("taco")))
+}
+
+func (t *CreateChunkUploaderTest) WrappedSucceeds() {
+	const name = "taco"
+	var err error
+
+	// Erase
+	ExpectCall(t.cache, "Erase")(Any())
+
+	// Wrapped
+	obj := &gcs.Object{
+		Name:       name,
+		Generation: 1234,
+	}
+
+	ExpectCall(t.wrapped, "CreateObject")(Any(), Any()).
+		WillOnce(Return(obj, nil))
+
+	// Insert
+	ExpectCall(t.cache, "Insert")(obj, timeutil.TimeEq(t.clock.Now().Add(ttl)))
+
+	// Call
+	o, err := t.bucket.CreateObject(context.TODO(), &gcs.CreateObjectRequest{})
```

Comment:
change CreateObject to CreateChunkUploader as we are not testing CreateObject here.

Reply:
I tried doing this, but this test needs the return values to be mocked, which is not possible for chunkUploader as it's an interface. In other words, a precreated instance of ChunkUploader needs to be passed here which can be updated within this call. 
From what I see, mocking is only possible for structs and primitives, but not for interface return-type values. This is probably why there are no tests for NewReader here as it returns io.Reader which is also an interface.
So, this wrapSucceeds test can't be implemented, unless we can find a way to mock ChunkUploader interface return-type.

Reply:
Disabling this test for now. Will remove the code once we have an agreement.

Reply:
Removed the test.